toolkit/tools/grapher/grapher_test.go

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package main
+
+import (
+	"os"
+	"testing"
+
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/logger"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/pkggraph"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/pkgjson"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestMain(m *testing.M) {
+	logger.InitStderrLog()
+	logger.Log.SetLevel(logrus.FatalLevel)
+	os.Exit(m.Run())
+}
+
+// Make a "local" package of the given name. Each package with the same pkgName will share the same
+// requires, srpmpath, etc. The subPkgName is used to differentiate multiple run-nodes for each
+// package. The buildRequires should be common across all packages with the same pkgName.
+func makePackage(pkgName, subPkgName string, buildRequires []*pkgjson.PackageVer) *pkgjson.Package {
+	// Create a package with the given name and version
+	pkg := &pkgjson.Package{
+		Provides: &pkgjson.PackageVer{
+			Name:    subPkgName,
+			Version: "1.0",
+		},
+		SrpmPath:     pkgName + ".src.rpm",
+		RpmPath:      pkgName + ".rpm",
+		SourceDir:    pkgName + "-src",
+		SpecPath:     pkgName + ".spec",
+		Architecture: "x86_64",
+		IsToolchain:  false,
+		RunTests:     true,
+	}
+
+	pkg.BuildRequires = buildRequires
```

Comment:
Irrelevant but just curious: why is `BuildRequires` set separately from all other members?

Reply:
Just cruft from a previous implementation